### PR TITLE
Add appimages to Linux build

### PR DESCRIPTION
**Pull request by jamboxboom**
Linux AppImages are now available.

### CODE REVIEW

**Comment by SecondFlight** [comment]
It looks like the build is working, and I can run the appimage with `--test` from the command line which means the build is being packaged properly with the Qt libraries.

However, it seems that the QML isn't being packaged correctly... in WinDeployQt, you tell it where the QML root directory is, and it then packages the QML files into the proper place along with whatever else is needed. It just sorta worked so I'm not sure about the details.

Here's what I get when I try running the appimage from the command line:
```
QQmlApplicationEngine failed to load component
qrc:/main.qml:4 module "QtGraphicalEffects" is not installed
qrc:/main.qml:2 module "QtQuick.Window" is not installed
qrc:/main.qml:5 module "QtQuick.Dialogs" is not installed
qrc:/main.qml:3 module "QtQuick.Shapes" is not installed
qrc:/main.qml:1 module "QtQuick" is not installed
qrc:/main.qml:4 module "QtGraphicalEffects" is not installed
qrc:/main.qml:2 module "QtQuick.Window" is not installed
qrc:/main.qml:5 module "QtQuick.Dialogs" is not installed
qrc:/main.qml:3 module "QtQuick.Shapes" is not installed
qrc:/main.qml:1 module "QtQuick" is not installed
qrc:/main.qml:4 module "QtGraphicalEffects" is not installed
qrc:/main.qml:2 module "QtQuick.Window" is not installed
qrc:/main.qml:5 module "QtQuick.Dialogs" is not installed
qrc:/main.qml:3 module "QtQuick.Shapes" is not installed
qrc:/main.qml:1 module "QtQuick" is not installed
qrc:/main.qml:4 module "QtGraphicalEffects" is not installed
qrc:/main.qml:2 module "QtQuick.Window" is not installed
qrc:/main.qml:5 module "QtQuick.Dialogs" is not installed
qrc:/main.qml:3 module "QtQuick.Shapes" is not installed
qrc:/main.qml:1 module "QtQuick" is not installed
qrc:/main.qml:4 module "QtGraphicalEffects" is not installed
qrc:/main.qml:2 module "QtQuick.Window" is not installed
qrc:/main.qml:5 module "QtQuick.Dialogs" is not installed
qrc:/main.qml:3 module "QtQuick.Shapes" is not installed
qrc:/main.qml:1 module "QtQuick" is not installed
```

This would seem to indicate that the QML itself is being packaged, but that it can't access the imports. I'm not completely sure what the issue is here.

@jamboxboom When you built the appimage, were you able to run it on your VM?

**Comment by jamboxboom** [comment]
Does the application use Qt Declarative / Qt Quick?

**Comment by SecondFlight** [comment]
Yes it does. As I understand, Qt Declarative is the name for Qt Quick 1 - this app uses Qt Quick 2. Dunno if that makes a difference.

**Comment by jamboxboom** [comment]
Naw it shouldn't. I have an idea on how to fix the problem based on this [guide](https://lemirep.wordpress.com/2013/06/01/deploying-qt-applications-on-linux-and-windows-3/). I'll look into it tonight.

File: anthem.desktop [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Type=Application
+Name=Anthem
+Comment=Better LMMS
```
Comment (SecondFlight) [inline anthem.desktop]: ok hol up

File: anthem.desktop [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+Exec=Main
+Icon=Anthem
+Categories=AudioVideo;
```
Comment (SecondFlight) [inline anthem.desktop]: Should this be `AudioVideo;Audio;`?

**Comment by SecondFlight** [comment]
> Naw it shouldn't. I have an idea on how to fix the problem based on this [guide](https://lemirep.wordpress.com/2013/06/01/deploying-qt-applications-on-linux-and-windows-3/). I'll look into it tonight.

I think it might actually matter. That article was written in 2013 before Qt Quick 2 came around, and Qt Quick 2 is a complete rewrite of Qt Quick/Qt Declarative.

It's not unlikely based on the error that a similar solution is in order, i.e. copying the right files to the right place in the appimage structure.

Here's an excerpt from the readme for `linuxdeploy/linuxdeploy-plugin-qt` which may help:
> ## Environment variables
> Just like all linuxdeploy plugins, the Qt plugin's behavior can be configured some environment variables.
> ...
> ### QML related:
> `$QML_SOURCES_PATHS`: directory containing the application's QML files -- useful/needed if QML files are "baked" into the binaries

I had to do a similar thing with `windeployqt` in specifying the QML directory, which is just the `Main` folder in the repo:
https://github.com/SecondFlight/daw-prototype/blob/7fe0b5383a10ee3e17522a2692a8da91da5c0fab/actions/windows/build-anthem.ps1#L9

So you should be able to do something similar, something like this (not 100% sure I got the syntax right):
```bash
export QML_SOURCES_PATHS=$GITHUB_WORKSPACE/Main
```

File: anthem.desktop (hunk 1 above)
Comment (SecondFlight) [inline anthem.desktop]: ok hol up

**Comment by jamboxboom** [comment]
### Tasks to Complete Before Merge

- [ ] Add Audio; to categories in anthem.desktop file
- [ ] Add QML libraries path to Linux environment variable $QML_SOURCES_PATHS

File: .github/workflows/build-linux.yml [hunk 3]
```diff
@@ -37,6 +37,7 @@ jobs:
       run: |
            cd $GITHUB_WORKSPACE/build
            export PATH=$PATH:/qt/5.13.2/gcc_64/bin/
+           export QML_SOURCES_PATHS=$GITHUB_WORKSPACE/build/Main
```
Comment (SecondFlight) [inline .github/workflows/build-linux.yml]: This shouldn't be set to anything in the build directory; it should just be wherever the `qml.qrc` file is, which is `(project root directory)/Main` in this case.

Comment (SecondFlight) [reply .github/workflows/build-linux.yml]: I'll make the change on web and see if it works...

**Pull request merged**